src/contextkit-subscriber/property.cpp

```diff
@@ -189,7 +189,9 @@ class UnsubscribeRequest : public Event
         , key_(key)
         , done_(std::move(done))
     {}
-    virtual ~UnsubscribeRequest() {}
+    virtual ~UnsubscribeRequest() {
+        done_.set_value();
+    }
 
     UnsubscribeRequest(SubscribeRequest const&) = delete;
```

Comment:
This is the future/promise pair (```on_unsubcribed_, done_```), so when request is deleted (after being processed or not) promise is marked as satisfied (```set_value()```) and future gets this as "ready". But ```wait_for()``` is not used during unsubscription to avoid "main" thread to be blocked. This promise should be satisfied only until ContextProperty is destroyed or resubscribed. I should document it.